src/uipath/_services/jobs_service.py

```diff
@@ -31,6 +42,284 @@ def __init__(self, config: Config, execution_context: ExecutionContext) -> None:
         self._temp_dir = os.path.join(tempfile.gettempdir(), TEMP_ATTACHMENTS_FOLDER)
         os.makedirs(self._temp_dir, exist_ok=True)
 
+    @traced(name="jobs_list", run_type="uipath")
+    def list(
+        self,
+        *,
+        folder_path: Optional[str] = None,
+        folder_key: Optional[str] = None,
+        filter: Optional[str] = None,
+        orderby: Optional[str] = None,
+        top: int = 100,
+        skip: int = 0,
+    ) -> Iterator[Job]:
+        """List jobs with auto-pagination.
+
+        Args:
+            folder_path: Folder path to filter jobs
+            folder_key: Folder key (mutually exclusive with folder_path)
+            filter: OData $filter expression
+            orderby: OData $orderby expression
+            top: Maximum items per page (default 100)
+            skip: Number of items to skip
+
+        Yields:
+            Job: Job instances
+
+        Examples:
+            >>> for job in sdk.jobs.list():
+            ...     print(job.key, job.state)
+        """
+        current_skip = skip
+
+        while True:
+            spec = self._list_spec(
+                folder_path=folder_path,
+                folder_key=folder_key,
+                filter=filter,
+                orderby=orderby,
+                skip=current_skip,
+                top=top,
+            )
+            response = self.request(
+                spec.method,
+                url=spec.endpoint,
+                params=spec.params,
+                headers=spec.headers,
+            ).json()
```

Comment:
Many orchestrator APIs also implement rate limiting with standard headers. The Get jobs API is limited to 100 calls per minute per tenant. We need to ensure that the sdk obeys the rate limit for all endpoints (as we may introduce more limits in the future).
https://docs.uipath.com/orchestrator/automation-cloud/latest/api-guide/rate-limits#exposed-headers